src/permutation.rs

```diff
@@ -37,22 +37,20 @@ impl Default for Permutation {
 }
 
 impl Permutation {
-    pub fn new(t: usize, full_rounds: usize, partial_rounds: usize) -> Result<Self, PermError> {
-        if full_rounds % 2 != 0 {
-            return Err(PermError::FullRoundsOdd);
-        }
-
-        let perm = Permutation {
+    // `full_rounds` input aims to be the number of full rounds that will be done
+    // before the partial rounds AND (BUT NOT ADDED TO) the number of full rounds 
+    // that will be done after the partial rounds.
+    pub fn new(t: usize, full_rounds: usize, partial_rounds: usize) -> Self {
+        
+        Permutation {
             t: t,
             full_rounds: full_rounds,
             partial_rounds: partial_rounds,
             data: Vec::with_capacity(t),
             data_lc: Vec::with_capacity(t),
             constants: RoundConstants::generate(full_rounds, partial_rounds, t),
```

Comment:
The round constants is currently expecting the `total_num_full_rounds`

Reply:
Saw it, now the hash tests are passing. 
Thanks for catching it!!